worker/genericworker/runmethod_windows.go

```diff
@@ -0,0 +1,68 @@
+package genericworker
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/taskcluster/taskcluster-worker-runner/protocol"
+	"github.com/taskcluster/taskcluster-worker-runner/run"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+func newServiceRunMethod() (runMethod, error) {
+	return &serviceRunMethod{}, nil
+}
+
+type serviceRunMethod struct {
+	serviceName string
+	mgr         *mgr.Mgr
+}
+
+func (m *serviceRunMethod) start(w *genericworker, state *run.State) (protocol.Transport, error) {
```

Comment:
Note, another option is to call `host.Run("net", "start", a.serviceName)`. See [Run](https://github.com/taskcluster/generic-worker/blob/a1c8307fa0095acdff6d8ccdb31a4a6bc7e2158c/host/host.go#L13-L17) method. One advantage of this is that we get logging for free, and if there is an issue starting up, it should be reasonably straightforward to reproduce via the CLI (i.e. we see in the logs the exact command called, and the stdout/stderr of the process). So overall it might be nicer as it is less code and a little more transparent in its activity, although it does mean executing a second process so might not be as efficient as use the native windows service library. Either way is probably fine though, I'm not sure myself which I prefer.

Reply:
I think the error handling here is pretty good -- in neither case would we see output from the service itself.  Also, the `wait` method needs to poll the service, so I think we need to use `svc` anyway -- might as well be consistent.

Reply:
Sorry, in the second I wrote the comment my head was inside generic-worker and I wasn't concentrating and thinking this was something running inside generic-worker, but of course this is the taskcluster-worker-runner side of the exchange. Shortly after writing the comment and shutting down the machine I suddenly realised that was of course code inside taskcluster-worker-runner and not generic-worker, so my comment made no sense. Sorry.

So yes, I agree - like you have it is fine, and I should have paid more attention. :-)

Reply:
I did notice that `net start generic-worker` somehow knows that generic-worker exited with code 73.  I can't find that information in `golang.org/x/sys/windows/svc`.  Any idea where it might be hiding?